core/src/main/java/org/infinispan/functional/impl/Params.java

```diff
@@ -108,4 +115,50 @@ public static Params from(Param<?>... ps) {
       }
    }
 
+   public static class Externalizer implements AdvancedExternalizer<Params> {
+      @Override
+      public Set<Class<? extends Params>> getTypeClasses() {
+         return Collections.singleton(Params.class);
+      }
+
+      @Override
+      public Integer getId() {
+         return Ids.PARAMS;
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, Params object) throws IOException {
+         for (Param param : object.params) {
+            switch (param.serializeMethod()) {
```

Comment:
@rvansa Param implementations will be provided by us, it's like flags really. Params control how the internals work, e.g. whether to check the cache store or not...etc, so really they are something we control. I don't expect users to implement Param instances. On the other hand, MetaParam instances can easily be provided by the user since they're stored along with the entry (like Metadata) so we have less control there.

I can see the utility of having a single externalizer for all Params, but without your extra method you'd require some kind of lookup. Besides, I don't think your method would work for Param instances that carry values.

One externalizer per Param would just work fine IMO.